ZcashFoundation/zcash-grant-system
# Email settings + unsubscribe

Closes #11 .

Feedback on email notification types welcome. 

### What this does
- adds `UserSettings` model with `email_subscriptions`
- adds "Email notifications" settings tab
- adds "Email unsubscribe" page using verification code (UX identical to verify)
- hooks up proposal approval emails + subscription check on `send_email`

### What this doesn't do
- hook up all email types for subscription checks
- fix generic email unsubscribe URL
- migrate existing users (w/o settings) you should reset your db

### Testing
- reset db
- create a user and visit settings > email notifications - test it out
- get verification code from the verify email, and visit `http://localhost:3000/email/unsubscribe?code=CODE`, should be successful
- go back to settings > email notifications - check that all are unchecked
- create and approve a proposal, you should not receive an email if unsubscribed from "Your Proposals - is approved or rejected"

## Screenshot
![screen shot 2019-01-11 at 6 06 27 pm](https://user-images.githubusercontent.com/11430954/51065936-25502880-15cd-11e9-88dd-70c481425a54.png)
